### PR TITLE
python311Packages.slither-analyzer: add pythonImportsCheck and passthru.tests

### PR DESCRIPTION
**Pull request by dotlambda**
## Description of changes
cc https://github.com/NixOS/nixpkgs/issues/294418

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/slither-analyzer/default.nix [hunk 1]
```diff
@@ -44,9 +46,38 @@ buildPythonPackage rec {
...
+    "slither.printers"
+    "slither.slither"
+    "slither.slithir"
+    "slither.solc_parsing"
+    "slither.utils"
+    "slither.visitors"
+    "slither.vyper_parsing"
+  ];
+
   # No Python tests
```
Comment (fabaff) [inline pkgs/development/python-modules/slither-analyzer/default.nix:71]: Looks like that there are [tests](https://github.com/crytic/slither/tree/master/tests) available.

**Comment by hellwolf** [comment]
@ofborg build slither-analyzer

File: pkgs/development/python-modules/slither-analyzer/default.nix (hunk 1 above)
Comment (dotlambda) [reply pkgs/development/python-modules/slither-analyzer/default.nix:71]: Not my beer.

**Comment by hellwolf** [comment]
The impor tests did not detect these regressions:

```
$ ./result/bin/slither --version
Traceback (most recent call last):
  File "/nix/store/f192lm98gx05v1awypjnp834qpckyfy0-python3.11-slither-analyzer-0.10.1/bin/.slither-wrapped", line 9, in <module>
    sys.exit(main())
             ^^^^^^
  File "/nix/store/f192lm98gx05v1awypjnp834qpckyfy0-python3.11-slither-analyzer-0.10.1/lib/python3.11/site-packages/slither/__main__.py", line 746, in main
    main_impl(all_detector_classes=detectors, all_printer_classes=printers)
  File "/nix/store/f192lm98gx05v1awypjnp834qpckyfy0-python3.11-slither-analyzer-0.10.1/lib/python3.11/site-packages/slither/__main__.py", line 760, in main_impl
    args = parse_args(all_detector_classes, all_printer_classes)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/f192lm98gx05v1awypjnp834qpckyfy0-python3.11-slither-analyzer-0.10.1/lib/python3.11/site-packages/slither/__main__.py", line 301, in parse_args
    version=require("slither-analyzer")[0].version,
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/ia0zlw106xb1d3vk9swyh8hgb0jfaycc-python3.11-setuptools-69.1.1/lib/python3.11/site-packages/pkg_resources/__init__.py", line 952, in require
    needed = self.resolve(parse_requirements(requirements))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/ia0zlw106xb1d3vk9swyh8hgb0jfaycc-python3.11-setuptools-69.1.1/lib/python3.11/site-packages/pkg_resources/__init__.py", line 813, in resolve
    dist = self._resolve_dist(
           ^^^^^^^^^^^^^^^^^^^
  File "/nix/store/ia0zlw106xb1d3vk9swyh8hgb0jfaycc-python3.11-setuptools-69.1.1/lib/python3.11/site-packages/pkg_resources/__init__.py", line 859, in _resolve_dist
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (eth-utils 4.0.0 (/nix/store/rs3rsrymslv2cs85fdyqcxhsj3rp0pl6-python3.11-eth-utils-4.0.0/lib/python3.11/site-packages), Requirement.parse('eth-utils<3,>=2.0.0'), {'rlp', 'eth-rlp', 'eth-keys', 'eth-keyfile'})
```

I am working on a PR right now. I will keep here updated.

**Comment by dotlambda** [comment]
@hellwolf See https://github.com/NixOS/nixpkgs/issues/294418.

**Pull request merged**